pkg/transport/transport.go

```diff
@@ -26,3 +26,27 @@ type Transport interface {
 	// In case slice is empty, the function should return an empty wrapped object.
 	WrapObjects(objects []*unstructured.Unstructured) runtime.Object
 }
+
+// TransportWithCreateOnly is a subtype of Transport that knows how to handle
+// the create-only option.
+type TransportWithCreateOnly interface {
+	Transport
+
+	// WrapObjectsHavingCreateOnly gets slice of Wrapee and wraps them into a single wrapped object.
+	// In case slice is empty, the function should return an empty wrapped object.
+	WrapObjectsHavingCreateOnly(objects []Wrapee) runtime.Object
```

Comment:
Does this function filter out non create-only wrapees or it expects the list to be of create-only enabled wrapees? The behavior is not clear.

Reply:
The intent is that this method honors the create-only bit, correctly handling either value.

Reply:
I pushed a revision that spells this out.

Reply:
Thanks. So this means that once the transition is complete, the name will revert back to `WrapObjects`?

Reply:
I was not planning on that but we can make that transition too with additional releases.